### DISCUSSION TITLE
Run automation tests in front of GitHub UI

### DISCUSSION THREAD

**Discussion by bcownerauto2** [community/community#38761] [Apps, API and Webhooks] [answered]
### Select Topic Area

Question

### Body

My company implemented a plugin for GitHub that scans repository code and warns about security violations.
I need to implement UI automation tests for this plugin so I manually created a dedicated user in GitHub and also created a repository with some files.
In my tests, I need to log in to GitHub via UI, which means navigating to https: https://github.com/, entering username and password, and clicking on the sign-in button.
Things work fine on my local machine but when trying to run it from Jenkins machine Github sends a device verification code to my Gmail account that connected to the GitHub user and the test failed.
This happens although I disabled the Two-factor authentication mechanism in my GitHub account.
I’m wondering what is the best way to solve this issue.
Please advice.

**Comment by airtower-luna** [comment, marked as answer]
> I need to implement UI automation tests for this plugin so I manually created a dedicated user in GitHub and also created a repository with some files.

Word of caution, I'm not sure if this is compatible with the [Account Terms section](https://docs.github.com/en/site-policy/github-terms/github-terms-of-service#3-account-requirements) of GitHub's ToS. Maybe the account could count as a machine account, if it's your only one for automation, but you might want to check the details before your account(s) get blocked. :sweat_smile: 

> This happens although I disabled the Two-factor authentication mechanism in my GitHub account.

At the technical level this is exactly the problem, because that's when GitHub sends those mails. :grin: If you set up 2FA with TOTP on the account and save the TOTP secret, you can generate the current TOTP code in your workflow and answer the TOTP challenge.
